Sources/Themes.php

```diff
@@ -879,6 +879,24 @@ function EnableTheme()
 	redirectexit('action=admin;area=theme;sa=list;' . $context['session_var'] . '=' . $context['session_id'] . ';done=' . (isset($_GET['disabled']) ? 'disabling' : 'enabling'));
 }
 
+/**
+ * Determines if a user can change their theme.
+ *
+ * @param int $id_member
+ * @param int $id_theme
+ *
+ * @return bool
+ */
+function canPickTheme($id_member, $id_theme)
+{
+	global $modSettings, $user_info;
+
+	return
+		allowedTo($user_info['id'] == $id_member ? 'profile_extra_own' : 'profile_extra_any')
+		&& (allowedTo('admin_forum') || $id_theme == 0 || in_array($id_theme, explode(',', $modSettings['knownThemes'])))
+		&& (!empty($modSettings['theme_allow']) || allowedTo('admin_forum'));
```

Comment:
I think it'd be good to also check that the theme isn't disabled, I was able to change to a theme that was disabled by the admin, although the layout remains as the default, as soon as the theme is enabled my layout changed to it.

Reply:
thhat's what the in_array on line 896 should do. I just tried it and it works. Am I missing something?

Reply:
I have a feeling that you, as admin, are changing the theme of another user.

Reply:
Well here's the thing, **$modSettings['knownThemes']** contains the themes which users are allowed to switch to, but **$modSettings['enableThemes']** contains the enabled themes, in 2.1 the ability to completely disable a theme was introduced, you can do that from the Theme Settings admin area, so right now **canPickTheme()** returns true with a normal user id & a disabled theme for parameters
![enableThemes](https://user-images.githubusercontent.com/20267363/55319822-0c417000-5476-11e9-9757-a70d9bb4db3c.png)